test/lsp-mode-test.el

```diff
@@ -102,5 +102,67 @@
   (should (string= "bar" (lsp-resolve-value 'lsp-test-my-var )))
   (should (string= "fn-result" (lsp-resolve-value (-const "fn-result" )))))
 
+(ert-deftest lsp-diagnostics-stats-test ()
+  (let ((workspace (make-lsp--workspace)))
+    (clrhash lsp-diagnostic-stats)
+    (lsp--on-diagnostics workspace
+                         (lsp-make-publish-diagnostics-params
+                          :uri (lsp--path-to-uri "/foo/bar/baz/txt.txt")
+                          :diagnostics (vector
+                                        (lsp-make-diagnostic :severity? lsp/diagnostic-severity-error)
+                                        (lsp-make-diagnostic :severity? lsp/diagnostic-severity-warning))))
+    (should (equal (lsp-diagnostics-stats-for "/foo")
```

Comment:
```suggestion
    (should (equal (lsp-diagnostics-stats-for (expand-file-name "/foo"))
```

This is needed since `lsp--path-to-uri` also expand to absolute path. In Windows, `/foo` is not absolute path yet